apis/addons/v1alpha1/addonoperators_types.go

```diff
@@ -27,6 +30,11 @@ type AddonOperatorSpec struct {
 	// +optional
 	Paused bool `json:"pause"`
 
+	// Send addon statuses to the addon service status API when set
+	// to true.
+	// +optional
+	EnableStatusReporting bool `json:"enableStatusReporting"`
```

Comment:
```suggestion
	EnableStatusReporting *bool `json:"enableStatusReporting"`
```

Shall we use pointers for this config as it will help us distinguish between explicit `false` and `not specified`.

note: usually recommended in kubeapi irrespective of type. [example from kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_types.go#L96)

Reply:
@nikhil-thomas In this case we want the implicit `not specified` to translate to false.

Reply:
True, however in the future if we want to make `not specified` to translate to true. There will be now way for our code to know whether the a false value was explicitly set or is it the default boolean val for golang bool type.

I think this will be a change that we can make in the future as well, as the API will not change.

Reply:
I wonder if there is a reason today to even make this a toggle in the API.
The Status API MUST be available in all environments and in the dev-env we already deploy the api-mock.

The only reason I can think of, why we would make this an API toggle, is to give our tenants the ability to use ADO for testing without OCM - But thats a ~story~ ~epic~ big rock for another day ;)

Not requiring this toggle would make this PR quite a lot less complicated

Reply:
@thetechnick  I was thinking of the following reasons for putting this feature behind a toggle :
- Decouple the availability of the status API with the promotion of this feature to stage/prod
- Like you said, allow folks to use the addon-operator w/o dependency on OCM/AS.
- Give us a mechanism(through the addon-operator CR) to disable status reporting fleet wide, in the event of some
    downtime in OCM/AS.
I've also opened a follow-up PR cleaning up these runtime toggles here: https://github.com/openshift/addon-operator/pull/294
Wdyt?

Reply:
@ashishmax31 
Points 1 and 2 could also be handled via a simple Environment variable, being way less complicated to implement than being able to toggle this feature during runtime.

For 3, unavailability of the OCM/AS API should not impact Addon Operator.
If this is really needed OCM should setup a central circuit breaker in their API middleware/LB to e.g. prevent requests in case of overload. AO is the wrong component to handle that particular issue.

Reply:
Agreed, lets remove this feature toggle then.